vars/wum-pipeline.groovy

```
@@ -0,0 +1,101 @@
+
+                          # Start running WUM script which have the whole procees of calling to WUM UAT and get the Products, Version and channel list
+                          mkdir wum_script
+                          cp -f ${SCENARIOS_LOCATION}/apim-test-integration/newshell.sh ${SCENARIOS_LOCATION}/wum_script
+                          cd ${SCENARIOS_LOCATION}/wum_script 
+                          sh get-wum-uat-products.sh
+                    
+                    """
+                }
+            }
+        }
+        stage('parallel-run') {
+            steps {
+                script {
+                    try{
+                        //Converting job name list text as a json file
+                        sh """
+                            jq -Rs '[ split("\n")[] | select(length > 0)
+                                | split(",") | {JobName: .[0]} ]' "${SCENARIOS_LOCATION}/wum_script/job-list.txt" > "${SCENARIOS_LOCATION}/wum_script/job-list.json"
+                         """
+                        def inputFile = new File("${SCENARIOS_LOCATION}/wum_script/job-list.json")
+                        def jsonSulper = new JsonSlurper()
+                        def InputJSON = jsonSulper.parseText(inputFile.text)
+                        jsonSulper = null
+
+                        def wumJobs = [:]
+
+                        //Get the job name from the job-list.json and trigger builds pararelly
+                        for (int index = 0; index < InputJSON.size(); index++) {
+                            String jobName = InputJSON[index].JobName;
+                            wumJobs["job:"+ jobName] = {build job: jobName};
+                        }
+                        InputJSON = null
+                        parallel wumJobs
+
+                        sh """
+                        set +e
+                        """
+                    }catch(e){
+                        echo "Few of the builds are not found to trigger."
```

actual exception is swallowed.